### PR TITLE
Cleanup ComputerName and UserName interop

### PR DESCRIPTION
**Pull request by hughbe**
## Proposed Changes
- Cleanup GetComputerName and make unicode
- Cleanup GetUserName and make unicode

## Test Methodology
Existing tests
```cs
        [Fact]
        public void SystemInformation_ComputerName_Get_ReturnsExpected()
        {
            string name = SystemInformation.ComputerName;
            Assert.InRange(name.Length, 1, 256);
            Assert.Equal(name, name.Trim('\0'));
            Assert.Equal(name, SystemInformation.ComputerName);
        }

        [Fact]
        public void SystemInformation_UserName_Get_ReturnsExpected()
        {
            string name = SystemInformation.UserName;
            Assert.InRange(name.Length, 1, 256);
            Assert.Equal(name, name.Trim('\0'));
            Assert.Equal(name, SystemInformation.UserName);
        }
```

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/2027)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/SystemInformation.cs [hunk 1]
```diff
@@ -619,13 +619,20 @@ public static bool MonitorsSameDisplayFormat
...
-        public static string ComputerName
+        public unsafe static string ComputerName
         {
             get
             {
-                var sb = new StringBuilder(256);
-                UnsafeNativeMethods.GetComputerName(sb, new int[] { sb.Capacity });
-                return sb.ToString();
+                const int BufferSize = 256;
+                Span<char> buffer = stackalloc char[BufferSize];
```
Comment (zsd4yr) [inline src/System.Windows.Forms/src/System/Windows/Forms/SystemInformation.cs]: I see you tested things; are you confident that the span works in place of the string builder here?

File: src/System.Windows.Forms/src/System/Windows/Forms/SystemInformation.cs [hunk 2]
```diff
@@ -619,13 +619,20 @@ public static bool MonitorsSameDisplayFormat
...
+                uint size = BufferSize;
+                fixed (char* pBuffer = buffer)
+                {
+                    Kernel32.GetComputerNameW(pBuffer, ref size);
```
Comment (danmosemsft) [inline src/System.Windows.Forms/src/System/Windows/Forms/SystemInformation.cs]: Just delete all this and call `Environment.MachineName` ?

File: src/Common/src/Interop/Advapi32/Interop.GetUserNameW.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Advapi32
+    {
+        [DllImport(Libraries.Advapi32, ExactSpelling = true)]
+        public unsafe static extern BOOL GetUserNameW(char* lpBuffer, ref uint pcbBuffer);
```
Comment (danmosemsft) [inline src/Common/src/Interop/Advapi32/Interop.GetUserNameW.cs]: Drop this and just call `Environment.UserName` ?

**Comment by hughbe** [comment]
@danmosemsft done!

**Comment by RussKie** [comment]
Nice one @danmosemsft! Thanks

**Pull request merged**